app/models/image_data.rb

```diff
@@ -47,4 +50,15 @@ def dimensions
                       Dimensions.new(image[:width], image[:height])
                     end
   end
+
+  def filename_is_unique
+    return if validate_on_edition.blank? || validate_on_image.blank? || file.blank?
+
+    edition = validate_on_edition
+    image = validate_on_image
+
+    if edition.images.excluding(image).any? { |i| i.filename == filename }
```

Comment:
Is this association already loaded in memory? Just wondering if it's potentially a big query we're running to load them all in which could be a query on the DB table with filename (that said I'm not sure where we get the filename from)